### ISSUE TITLE
Consider changing `add_edge(src, dst, e)` to `add_edge(e: impl IntoEdge)`

### ISSUE THREAD

**Issue by pnevyk** [pnevyk/gryf#51]
Advantages:

* Cleaner (?) API for unweighted graphs: `graph.add_edge((src, dst))` instead of `graph.add_edge(src, dst, ())`.

Disadvantages:

* Worse documentation? `add_edge(&mut self, src: VertexIndex, dst: VertexIndex, edge: E)` is much clearer than `add_edge<T: IntoEdge<VertexIndex, E, Ty>>(&mut self, edge: T)`.
* Potential problems with type inference. Especially in case of encapsulations where we don't use `G::VertexIndex` directly, but `VI: Borrow<G::VertexIndex>` instead.

---

In summary, I am not much in favor of this change, but just wanted to leave a note if I change the opinion and the disadvantages will turn out not that bad.
